app/users/handlers.js

```diff
@@ -27,13 +27,24 @@ exports.user_details = (req, res, next) => {
 
 
 exports.user_edit = (req, res, next) => {
-  Promise.all([User.get(req.params.id), App.list(), Bucket.list()])
-    .then(([user, apps, buckets]) => {
-      res.render('users/edit.html', {
-        user,
-        apps_options: apps.exclude(user.apps),
-        buckets_options: buckets.exclude(user.buckets),
-      });
+  User.get(req.params.id)
+    .then((user) => {
+      user.is_superuser = !!req.body.superadmin;
+
+      return user.update()
+        .then(() => {
+          req.session.flash_messages.push('Updated user');
+
+          let url = url_for('users.details', { id: user.auth0_id });
+
+          if (user.auth0_id === req.user.auth0_id) {
+            // force re-login
+            url = url_for('base.logout');
```

Comment:
Why do we force re-login? I imagine the frontend app caches the user data somewhere so that would need updating but is there any other reason why users would need to re-login if they update themselves? Does it has anything to do with tokens?

Reply:
If you remove your own superuser status, you are forced to re-login, as you are still a superuser in your current session. I could modify the user object in the session instead, and redirect to the home page, which would achieve the same sort of thing without having to login again, I suppose.

Reply:
Yeah, that's what I meant. Unless the token needs to change or any of the other fields of the user object is not coming from the response and it's invalidate by this request or something like that.